### DISCUSSION TITLE
Is there an MQTT topic to get robot information (Manufacturer, model, versions, etc?)

### DISCUSSION THREAD

**Discussion by awefdev** [Hypfer/Valetudo#1965] [Q&A / Support]
I pulled down the code and I see it as a HASS attribute, but I don't see it as an MQTT topic (even with Homie support enabled). Is there anything like this? And if not, would you be willing to accept a PR to include it?

**Comment by Hypfer** [comment]
No there is no such MQTT Topic. What would be the purpose of it? What use-case would you like to solve?

**Comment by awefdev** [comment]
I'm creating a small React Native application to control a Valetudo vacuum and I would like a way to display a picture of the robot selected, both for visual flair and for helping users with multiple vacuum models to have another visual cue which they're working with. Roborock's app does a similar thing.

**Reply by Hypfer** [reply]
I can't stop you from doing it and if you're just looking for a fun challenge then there's nothing wrong with that but in general, I would be happy to not see that app appear as something intended to be used by other people.
This is because recreating the same functionality that already exists will likely introduce new bugs and inconsistencies and that will only reflect badly on the project. 

The cool thing with Valetudo providing its own interface bundled with it as a PWA is that the frontend version will always exactly match the backend version and thus will also only need code that expects that specific backend version. If you decouple those things then suddenly both need to try to support a whole range of versions of the other which too leads to unexpected bugs and general worse UX.

Being a native app also then requires you to be on the stores and keep that up to date and deal with all the nonsense the cloud landlords throw at you.
And at the end of the day, you don't really gain anything, because the PWA works great as a PWA. I guess you could squeeze out a bit more performance from native controls but since performance hasn't been an issue previously.. eh.

Then again, I have no authority to tell you what you should do in your free time. I'm just saying that there are very good reasons why Valetudos Frontend is a PWA and not a native app.

**Reply by awefdev** [reply]
That's completely understandable, and no I'd never release this without broader community involvement first, but my issue with PWAs is that they seem to be sort of a half-way house solution. You're stuck with browser capabilities and limitations which aren't there in native applications since you have more control-- though I fully admit this affects UX more than usability. I actually first tried this a while back with a PWA but ended up running into weird bugs, especially when trying to make it work across browsers + iOS.

**Reply by Hypfer** [reply]
But are you aware of any specific cases where being a PWA limits the Valetudo UX?

The only thing I could think of would be the APIs that are only available in secure contexts but the only thing in there I can think of would be the ability to send push notifications and tbh I probably wouldn't want those even if I could.

**Reply by awefdev** [reply]
Push notifications was actually one of the first things come to mind, since I can see it potentially beneficial for alerting to problems or giving a quick vacuum status. There's also the stubbed out address bar at the top of the screen, as well as it having to load via the browser loading bar before it actually loads into the PWA.

But I'm in no way saying Valetudo being a PWA isn't the absolute right choice. I'm not trying to replace Valetudo Companion or something, it's just a small app based around ease of selecting rooms (from the MapSegments topic) and pre-saved zones. I don't see how this is any different than a custom Home Assistant card controlling via MQTT?
